src/client.rs

```diff
@@ -323,6 +324,31 @@ impl Client {
         }
     }
 
+    /// This deletes a resource _if it is not deleted already_ and waits until the deletion is
+    /// performed by Kubernetes.
+    ///
+    /// It calls `delete` to perform the deletion.
+    ///
+    /// Afterwards it loops and checks regularly whether the resource has been deleted
+    /// from Kubernetes
+    pub async fn ensure_deleted<T>(&self, resource: T) -> OperatorResult<()>
```

Comment:
Due to the nature of finalizers I believe we need a timeout here.

Reply:
I considered that, but we don't really offer timeouts in many other places and instead let users wrap the call in a timeout themselves if they need it.
I figured this was more in line with other fn calls - but am happy to add it too.
No real preference..

Reply:
Fair enough